packages/medusa-payment-stripe/src/subscribers/cart.js

```diff
@@ -1,53 +1,58 @@
 class CartSubscriber {
   constructor({
+    manager,
     cartService,
-    customerService,
     paymentProviderService,
     eventBusService,
   }) {
     this.cartService_ = cartService
-    this.customerService_ = customerService
     this.paymentProviderService_ = paymentProviderService
     this.eventBus_ = eventBusService
+    this.manager_ = manager
 
     this.eventBus_.subscribe("cart.customer_updated", async (cart) => {
       await this.onCustomerUpdated(cart)
     })
   }
 
   async onCustomerUpdated(cartId) {
-    const cart = await this.cartService_.retrieve(cartId, {
-      select: [
-        "subtotal",
-        "tax_total",
-        "shipping_total",
-        "discount_total",
-        "total",
-      ],
-      relations: [
-        "items",
-        "billing_address",
-        "shipping_address",
-        "region",
-        "region.payment_providers",
-        "items",
-        "items.adjustments",
-        "payment_sessions",
-        "customer",
-      ],
-    })
+    await this.manager_.transaction(async (transactionManager) => {
+      const cart = await this.cartService_
+        .withTransaction(transactionManager)
+        .retrieve(cartId, {
+          select: [
+            "subtotal",
+            "tax_total",
+            "shipping_total",
+            "discount_total",
+            "total",
+          ],
+          relations: [
```

Comment:
@oliverheaps should we pass those as well ?
```
"discounts",
"discounts.rule",
```

Cause we are requesting the discount total as well

Reply:
maybe that we also need the gift_card to be taken into account? for the total and the relations

Reply:
We are passing `items` and `items.adjustments` which should be sufficient to get the `discount_total`. 

Yes, I think you are right. We need to add `gift_card_total` to the select query.